homeassistant/components/zwave/__init__.py

```diff
@@ -297,15 +297,46 @@ def value_added(node, value):
     def node_added(node):
         """Handle a new node on the network."""
         entity = ZWaveNodeEntity(node, network)
-        name = node_name(node)
-        generated_id = generate_entity_id(DOMAIN + '.{}', name, [])
-        node_config = device_config.get(generated_id)
-        if node_config.get(CONF_IGNORED):
-            _LOGGER.info(
-                "Ignoring node entity %s due to device settings",
-                generated_id)
+
+        def _add_node_to_component():
+            name = node_name(node)
+            generated_id = generate_entity_id(DOMAIN + '.{}', name, [])
+            node_config = device_config.get(generated_id)
```

Comment:
(I realize it's existing logic)

This is not entirely correct anymore, as based on the unique ID we might choose to give them a different entity id. The correct implementation would be to leverage the disabling of entities via the entity registry. If an entity is disabled, we will never call `async_added_to_hass` (where subscriptions should happen)

Reply:
Yes, this bug is what got me working on zwave again in the first place.

I'll will address this in a future PR.

In any case, is the registry the right place for ignoring entities? Do other platforms use it for "ignore"?

Also - imaging a zombie zwave node that can't be parsed at all. This means it won't get unique_id, so it can't be ignored via the registry.

I propose to add a a list of ignored IDs to (root) zwave config to handle this case.


As for ignoring specific values - while entity registry would work here, I find it very convenient to ignore via zwave config globs, i.e:
```
sensor.*energy*:
  ignored: true
```

Reply:
Btw a useful subset of globs can be used from the UI - a checkbox "ignore energy meters", "ignore power meters", etc.